plugins/inputs/logfile/logfile.go

```diff
@@ -107,7 +107,8 @@ func (t *LogFile) Start(acc telegraf.Accumulator) error {
 		return fmt.Errorf("failed to create state file directory %s: %v", t.FileStateFolder, err)
 	}
 
-	// Clean state file regularly
+	// Clean state file on init and regularly
+	t.cleanupStateFolder()
```

Comment:
The agent reads the state to track which part of the log file it left off reading. Does this execute before or after the state file has been read?

Reply:
nit: Move this into the goroutine. I don't think we need to block on the clean up.